torchqdynamics/solver.py

```diff
@@ -104,11 +104,20 @@ def save(self, y: Tensor):
         self._save_exp_ops(y)
         self.save_counter += 1
 
+    def save_initial(self, y: Tensor):
+        if self.options.save_states and self.next_tsave() == 0.0:
+            self.save(y)
```

Comment:
This logic is very specific to the adaptive ODE solver, so it should be close to the adaptive solver code, not in the very general `Solver` base class. I suggest removing this function and replace in `_adaptive_odeint`:
```diff
# save initial solution
- self.save_initial(self.y0)
+ if self.next_tsave() == 0.0:
+    self.save(y)
```